esp/services/espcontrol/ws_espcontrolservice.cpp

```diff
@@ -75,6 +75,104 @@ void CWSESPControlEx::init(IPropertyTree *cfg, const char *process, const char *
     }
 }
 
+bool CWSESPControlEx::onDetachFromDali(IEspContext& context, IEspDetachFromDaliRequest& req, IEspDetachFromDaliResponse& resp)
+{
+    int status = 0;
+    bool force = req.getForce_isNull() ? false : req.getForce();
+    VStringBuffer message("Request to DETACH ESP Process '%s' from Dali ", this->espProcess.get());
+    CEspServer * thisESPServer = static_cast<CEspServer *>(m_container);
+    if (thisESPServer)
+    {
+        message.append("has been issued - ");
+        if (thisESPServer->detachESPFromDali(force))
+            message.append("and success reported.");
+        else
+        {
+            message.append("but failure reported.");
+            status = -1;
+        }
+    }
+    else
+    {
+	message.append("COULD NOT BE ISSUED due to internal error");
+	status = -1;
+    }
+    resp.setMessage(message.str());
+    resp.setStatus(status);
+    return status;
+}
+
+bool CWSESPControlEx::onAttachToDali(IEspContext& context, IEspAttachToDaliRequest& req, IEspAttachToDaliResponse& resp)
+{
+    int status = 0;
+    VStringBuffer message("Request to ATTACH ESP Process '%s' to Dali ", this->espProcess.get());
+    CEspServer * thisESPServer = static_cast<CEspServer *>(m_container);
+    if (thisESPServer)
+    {
+	message.append("has been issued - ");
+        if (thisESPServer->attachESPToDali())
+            message.append("and success reported.");
+        else
+        {
+            message.append("but failure reported.");
+            status = -1;
+        }
+    }
+    else
+    {
+        message.append("COULD NOT BE ISSUED due to internal error");
+        status = -1;
+    }
+    resp.setMessage(message.str());
+    resp.setStatus(status);
+    return status;
+}
+
+bool CWSESPControlEx::onUnsubscribeFromDali(IEspContext& context, IEspUnsubscribeFromDaliRequest& req, IEspUnsubscribeFromDaliResponse& resp)
+{
```

Comment:
Same comments as above... why this-> and could common up code.

Reply:
this-> usually comes about from utilizing intellisense to find member vars, it makes it to the pull req because it helps identifying member vars without any other context. some might see it as noise, but it can actually be helpful sans IDE colorization/etc.